src/NuGet.Clients/PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -211,21 +218,39 @@ private void LoadItems(PackageItemListViewModel selectedPackageItem)
             token.ThrowIfCancellationRequested();
         }
 
-        private async Task<IEnumerable<PackageItemListViewModel>> LoadNextPageAsync(IItemLoader<PackageItemListViewModel> currentLoader, CancellationToken token)
+        private async Task<IEnumerable<PackageItemListViewModel>> LoadNextPageAsync(PackageItemLoader currentLoader, CancellationToken token)
         {
             var progress = new Progress<IItemLoaderState>(
                 s => HandleItemLoaderStateChange(currentLoader, s));
 
-            // trigger loading
-            await currentLoader.LoadNextAsync(progress, token);
+            // if searchResultTask is in progress then just wait for it to complete
+            // without creating new load task
+            if (_initialSearchResultTask != null)
+            {
+                // update initial progress
+                var cleanState = SearchResult.Empty<IPackageSearchMetadata>();
+                await currentLoader.UpdateStateAndReportAsync(cleanState, progress, token);
+
+                var results = await _initialSearchResultTask;
+
+                // update state and progress
+                await currentLoader.UpdateStateAndReportAsync(results, progress, token);
 
-            // run till first results are ready
-            for (var state = currentLoader.State;
-                state.LoadingStatus == LoadingStatus.Loading && state.ItemsCount == 0;
-                state = currentLoader.State)
+                _initialSearchResultTask = null;
+            }
+            else
             {
-                token.ThrowIfCancellationRequested();
-                await currentLoader.UpdateStateAsync(progress, token);
+                // trigger loading
+                await currentLoader.LoadNextAsync(progress, token);
+
+                // run till first results are ready
+                for (var state = currentLoader.State;
+                    state.LoadingStatus == LoadingStatus.Loading && state.ItemsCount == 0;
+                    state = currentLoader.State)
```

Comment:
I know this is the same as before, but it's really confusing that this is a for loop instead of a while loop